### DISCUSSION TITLE
Inconsistent API Responses for YouTube License Validation – Need Clarification

### DISCUSSION THREAD

**Discussion by rajnishdargan** [Sunbird-Knowlg/Community#223] [Q&A] [answered]
Hello Team,

I’m encountering an inconsistency with the response from the asset license validation API when hitting it directly from content-service and when hitting it through portal.

**Note** - We are using knowlg-inQuiry service setup using [sunbird installer release-7.0.0](https://github.com/project-sunbird/sunbird-ed-installer/tree/release-7.0.0)

Here's the issue:

When I hit this API on the content service:
`POST content_service_url/asset/v4/validate?field=license`

**Payload:**
```
{
  "request": {
    "asset": {
      "provider": "youtube",
      "url": "https://www.youtube.com/watch?v=9xWrG4xsHuc"
    }
  }
}
```
**Response:**
```
{
  "id": "asset.url.validate",
  "ver": "4.0",
  "ts": "2024-10-14T13:05:10ZZ",
  "params": {
    "resmsgid": "e4728db0-f2d1-4139-96a4-1bdb75b6a194",
    "msgid": null,
    "err": "ERR_YOUTUBE_LICENSE_VALIDATION",
    "status": "failed",
    "errmsg": "Please Provide Valid YouTube URL!"
  },
  "responseCode": "CLIENT_ERROR",
   "result": {
    "messages": null
  }
}
```
However, in the portal, when hitting this similar API:
`POST {{host}}/action/asset/v3/validate?field=license`

**Response:**
```
{
  "id": "asset.url.validate",
  "ver": "1.0",
  "ts": "2024-10-14T13:29:29ZZ",
  "params": {
    "resmsgid": "710dbbe4-36d3-4636-b9d7-78632a2b8ded",
    "msgid": null,
    "err": null,
    "status": null,
    "errmsg": null
  },
  "responseCode": "OK",
  "result": {
    "license": {
      "valid": false,
      "value": "youtube"
    }
  }
}
```

As you can see, in the first case (content service), I am getting an error `ERR_YOUTUBE_LICENSE_VALIDATION` with `CLIENT_ERROR` status, whereas in the portal, I get a proper validation response without any errors and the status OK.

I wanted to understand why there is this discrepancy between the two responses. Is this something handled by middleware, or is there a difference in how the APIs are being processed?

Any insights would be greatly appreciated!

@pallakartheekreddy @HarishGangula 
@coolbung @itsvick @snehal0904 @arati-tekdi

**Comment by pallakartheekreddy** [comment, marked as answer]
@rajnishdargan Yes, we have the `knowledge-mw-service`, which will handle this
